Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentModeBus.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/ComponentBus.h>
+#include <Editor/Source/ComponentModes/Joints/JointsComponentModeCommon.h>
+
+namespace PhysX
+{
+    //! Request bus for Joints ComponentMode operations.
+    class JointsComponentModeRequests : public AZ::EntityComponentBus
+    {
+    public:
+        //! Set the current sub-mode.
+        virtual void SetCurrentSubMode(JointsComponentModeCommon::SubComponentModes::ModeType newMode) = 0;
+
+        //! Reset the current sub-mode.
+        virtual void ResetCurrentSubMode() = 0;
+
+        //! Returns true if the sub-mode should be available for the current component.
+        virtual bool IsCurrentSubModeAvailable(JointsComponentModeCommon::SubComponentModes::ModeType mode) = 0;
```

Comment:
nit: This could be made `const` (I realize it's a bus but the `const` is still a useful indicator for people looking at and implementing the API (there's always `mutable` if needed)